### PR TITLE
Smart Serialization for sourcemaps

### PR DESCRIPTION
**Pull request by klokane**
Automaticaly full rendering for elements with "SourceMaps"

### CODE REVIEW

File: src/refract/SerializeVisitor.cc [hunk 1]
```diff
@@ -54,6 +55,23 @@ namespace refract
             return array;
         }
 
+        IElement::renderFlags SelectSerializationType(const IElement& e, bool exportSourceMap) {
+
+            if (exportSourceMap) {
+                IElement::MemberElementCollection::const_iterator it = e.attributes.find("sourceMap");
+                // there is sourceMap in attributes
```
Comment (kylef) [inline src/refract/SerializeVisitor.cc]: Btw, an "objective-c-ism" to turn this comment into a property:

```cpp
bool attributesIncludesSourceMap = it != e.attributes.end();

if (attributesIncludesSourceMap) {
    IElement::rFull
}
```

I think this is a great way to write more readable code without comments.

**Comment by kylef** [comment]
This looks good to me, but Circle CI is failing because this introduces a build warning:

```
../src/refract/SerializeCompactVisitor.cc:19:84: error: type qualifiers ignored on function return type [-Werror=ignored-qualifiers]
         const bool operator()(const IElement* element, const bool exportSourceMap) const {
```

**Comment by klokane** [comment]
ahh, will fix

**Comment by klokane** [comment]
Fixed, push forced

**Pull request merged**